projects/hathor-wallet-headless/api-docs-ci.md

```diff
@@ -0,0 +1,68 @@
+# Summary
+This feature automates the process of uploading the updated API Docs for the Headless Wallet, while giving the developer a preview of the contents, error checking and linting.
+
+A script is made available for the developer to make all the necessary checks locally and a GitHub Workflow is implemented to:
+- Validate and lint the `api-docs.js` file on every PR to the `dev` branch
+- Deploy the updated docs on every new successful release
+
+# Motivation
+Currently this upload is made manually by a team member, extracting data from the `src/api-docs.js` file and processing it to finally upload its contents to the [Headless Hathor Wallet API](https://wallet-headless.docs.hathor.network/) site.
+
+# Guide-level explanation
+### JS to JSON conversion
+The platform that manages the OpenAPI Documentation static website, [Redoc](https://github.com/Redocly/redoc), only interacts with `.json` files. So we need to convert our `src/api-docs.js` file into a `.json` one whenever we want to use this platform.
+
+A script should be built to make this conversion on the `scripts` folder, that implements the following:
+- Ensures the `default` property normally exported by the `JSON.stringify()` is not present on the output json
+- Outputs the results to the `tmp/api-docs.json` folder.
+
+This script will be referred to as `scripts/convert-docs.js` from now on.
+
+> *Note*: The `securitySchemes` from the `components` property is no longer necessary and can be removed from the `js` file when this is implemented.
+
+### Linting and Preview
+A developer should be able to run a `npm run docs` command with three outcomes:
+- A validation check should be executed on the generated docs, alerting any errors found
+- A lint should be executed on the generated docs, displaying its analysis results
+- A local server should be fired, allowing the developer to visualize the results of the doc changes
+
+This would be an automated version of manually executing the following commands:
+```sh
+# Validation and conversion
+node ./scripts/convert-docs.js
+
+# Linting
+npx @redocly/cli lint tmp/api-docs.json
+
+# Local server for validating
+npx @redocly/cli preview-docs tmp/api-docs.json
+xdg-open http://127.0.0.1:8080
```

Comment:
Why do you need this? The redoc-cli can generate an html file, then you can just `open index.html` and the developer can see the generated file in the browser.

Better than adding another dependency in my opinion, even if it's a dev dependency

Reply:
`xdg-open` is the Ubuntu version for Mac's `open` command. So, no dependency will be added: it's only a request to open the browser on the local server that the cli starts.

The only dependency that will be added here is, indeed, the redoc-cli ( named [`@redocly/cli`](https://redocly.com/docs/cli/) )